### PR TITLE
treewide: originalName -> packPathName; luaName -> moduleName

### PR DESCRIPTION
**Pull request by khaneliman**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -73,12 +73,12 @@ A template plugin can be found in (plugins/TEMPLATE.nix)[https://github.com/nix-
...
 | **isColorscheme**            | Indicating whether the plugin is a colorscheme.                                                                                                                                                                                                      | No       | `false`                                                                                         |
-| **luaName**                  | The Lua name for the plugin.                                                                                                                                                                                                                         | No       | `name` parameter                                                                                |
+| **moduleName**                  | The Lua name for the plugin.                                                                                                                                                                                                                         | No       | `name` parameter                                                                                |
```
Comment (MattSturgeon) [inline CONTRIBUTING.md:76]: Unintended whitespace changes; need to re-align table columns

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -73,12 +73,12 @@ A template plugin can be found in (plugins/TEMPLATE.nix)[https://github.com/nix-
...
-| **originalName**             | The original name of the plugin, usually the plugin's github repo name. E.g. `"foo-bar.nvim"`.                                                                                                                                                       | No       | `name` parameter                                                                                |
+| **packPathName**             | The name of the plugin directory in [packpath](https://neovim.io/doc/user/options.html#'packpath'), usually the plugin's github repo name. E.g. `"foo-bar.nvim"`.                                                                                                                                                       | No       | `name` parameter                                                                                |
```
Comment (MattSturgeon) [inline CONTRIBUTING.md:79]: Same

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -73,12 +73,12 @@ A template plugin can be found in (plugins/TEMPLATE.nix)[https://github.com/nix-
...
 | **package**                  | The nixpkgs package attr for this plugin. Can be a string, a list of strings, a module option, or any derivation. For example, "foo-bar-nvim" for `pkgs.vimPlugins.foo-bar-nvim`, or `[ "hello" "world" ]` will be referenced as `pkgs.hello.world`. | No       | `name` parameter                                                                                |
-| **settingsDescription**      | A description of the settings provided to the `setup` function.                                                                                                                                                                                      | No       | `"Options provided to the require('${luaName}')${setup} function."`                             |
+| **settingsDescription**      | A description of the settings provided to the `setup` function.                                                                                                                                                                                      | No       | `"Options provided to the require('${moduleName}')${setup} function."`                             |
```
Comment (MattSturgeon) [inline CONTRIBUTING.md:81]: Same

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -135,7 +135,7 @@ Such plugins are usually configured via vim globals, but often have no configura
...
 | **optionsRenamedToSettings** | List of options renamed to settings.                                                                                                                                                                                                               | No       | `[]`                                  |
-| **originalName**             | The original name of the plugin, usually the plugin's github repo name. E.g. `"foo-bar.vim"`.                                                                                                                                                      | No       | `name` parameter                      |
+| **packPathName**             | The name of the plugin directory in [packpath](https://neovim.io/doc/user/options.html#'packpath'), usually the plugin's github repo name. E.g. `"foo-bar.vim"`.                                                                                                                                                      | No       | `name` parameter                      |
```
Comment (MattSturgeon) [inline CONTRIBUTING.md:138]: Same

File: lib/options.nix [hunk 5]
```diff
@@ -335,10 +335,10 @@ rec {
     };
 
   mkLazyLoadOption =
-    originalName:
+    packPathName:
```
Comment (MattSturgeon) [inline lib/options.nix:338]: In this context, we can probably just name it `name`

File: plugins/TEMPLATE.nix [hunk 6]
```diff
@@ -4,7 +4,7 @@ let
 in
 lib.nixvim.neovim-plugin.mkNeovimPlugin {
   name = "my-plugin";
-  originalName = "my-plugin.nvim"; # TODO replace (or remove entirely if it is the same as `name`)
+  packPathName = "my-plugin.nvim"; # TODO replace (or remove entirely if it is the same as `name`)
```
Comment (MattSturgeon) [inline plugins/TEMPLATE.nix:7]: Separate commit: maybe it'd be worth also having `moduleName` in this template?

File: plugins/by-name/lsp-signature/default.nix [hunk 7]
```diff
@@ -40,7 +40,7 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
       Will show two lines of comment/doc(if there are more than two lines in the doc, will be truncated).
 
       Set to zero if you DO NOT want any API comments be shown.
-      This setting only take effect in insert mode, it does not affect signature help in nornal mode.
+      This setting only take effect in insert mode, it does not affect signature help in normal mode.
```
Comment (MattSturgeon) [inline plugins/by-name/lsp-signature/default.nix:43]: Unrelated change: is this in a separate commit?
Or was this required by the pre-commit hook? (Is so I don't mind having it in one commit)

File: plugins/by-name/lsp-signature/default.nix [hunk 8]
```diff
@@ -189,7 +189,7 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
     # TODO: figure out the types
     keymaps = defaultNullOpts.mkListOf lib.types.anything [ ] ''
       Related to move_cursor_key. 
-      The keymaps inside floating window with arguements of bufnr. 
+      The keymaps inside floating window with arguments of bufnr.
```
Comment (MattSturgeon) [inline plugins/by-name/lsp-signature/default.nix:192]: Same

**Review by MattSturgeon — COMMENTED** [review]
I've looked at all the files, no real issues. Just a couple things worth briefly discussing.

File: plugins/by-name/lsp-signature/default.nix (hunk 7 above)
Comment (khaneliman) [reply plugins/by-name/lsp-signature/default.nix:43]: Sorry, out pre-commit hook prevented my change from committing because of this.

**Comment by khaneliman** [comment]
lol had to toggle markview, made it hard to see what the actual markdown looked like for tables...

File: CONTRIBUTING.md (hunk 2 above)
Comment (MattSturgeon) [reply CONTRIBUTING.md:80]: Still additional whitespace after the desc. Almost looks like you have a vim conceal level enabled and the hyperlink syntax is being hidden visually while you're checking alignment?

File: CONTRIBUTING.md (hunk 4 above)
Comment (MattSturgeon) [reply CONTRIBUTING.md:139]: Same as https://github.com/nix-community/nixvim/pull/2658/files#r1884754534

File: CONTRIBUTING.md (hunk 2 above)
Comment (khaneliman) [reply CONTRIBUTING.md]: Sorry, didn't see the stuff that far over :P

**Comment by khaneliman** [comment]
Sorry, didn't notice everything got shifted, I think I got it this time

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -73,12 +73,12 @@ A template plugin can be found in (plugins/TEMPLATE.nix)[https://github.com/nix-
...
 | **optionsRenamedToSettings** | Options that have been renamed and move to the `settings` attribute.                                                                                                                                                                                 | No       | `[]`                                                                                            |
-| **originalName**             | The original name of the plugin, usually the plugin's github repo name. E.g. `"foo-bar.nvim"`.                                                                                                                                                       | No       | `name` parameter                                                                                |
+| **packPathName**             | The name of the plugin directory in [packpath](https://neovim.io/doc/user/options.html#'packpath'), usually the plugin's github repo name. E.g. `"foo-bar.nvim"`.                                                                                                                                                       | No       | `name` parameter             |
```
Comment (MattSturgeon) [inline CONTRIBUTING.md:79]: I think this is the right alignment?

```suggestion
| **packPathName**             | The name of the plugin directory in [packpath](https://neovim.io/doc/user/options.html#'packpath'), usually the plugin's github repo name. E.g. `"foo-bar.nvim"`.                                                                                    | No       | `name` parameter                                                                                |
```

Comment (MattSturgeon) [reply CONTRIBUTING.md:79]: The other lines all look good

Comment (khaneliman) [reply CONTRIBUTING.md]: lol damnit

Comment (khaneliman) [reply CONTRIBUTING.md]: > (This is why I hate visually aligned md tables and proposed we not even attempt alignment in the user-configs doc)

Yeah, that table is WIDE. Makes it hard to glance and see it.

**Review by MattSturgeon — APPROVED** [review]
Nice work!

**Comment by khaneliman** [comment]
@mergify queue
